app/src/main/java/com/example/sharewhatyoucanproject/LoginActivity.kt

```diff
@@ -62,170 +56,155 @@ class LoginActivity : AppCompatActivity() {
-                                .addOnCompleteListener(OnCompleteListener { task3 ->
-                                    if (task3.isSuccessful) {
-                                        pd.dismiss()
-                                        if (type == 1) {
-                                            myEdit.putInt("selector", 1)
-                                            myEdit.commit()
-                                            val i = Intent(
-                                                this@LoginActivity,
-                                                SelectUserActivity::class.java
-                                            )
-                                            startActivity(i)
-                                            finishAffinity()
-                                        } else if (type == 2) {
-                                            myEdit.putInt("selector", 2)
-                                            myEdit.commit()
-                                            val i = Intent(
-                                                this,
-                                                DrawerActivity2::class.java
-                                            )
-                                            startActivity(i)
-                                            finishAffinity()
-                                        }
-                                    } else {
-                                        pd.dismiss()
-                                        Toast.makeText(
-                                            applicationContext,
-                                            "Failed " + task3.exception,
-                                            Toast.LENGTH_SHORT
-                                        ).show()
-                                    }
-                                })
-                        } else {
-                            pd.dismiss()
-                            Toast.makeText(
-                                applicationContext,
-                                "Failed " + task2.exception,
-                                Toast.LENGTH_SHORT
-                            )
-                                .show()
-                        }
+        ).addOnCompleteListener(
+            OnCompleteListener {
+                if (it.isSuccessful) {
+                    pd.dismiss()
+                    Toast.makeText(applicationContext, "Created", Toast.LENGTH_SHORT).show()
+                    val usermap: MutableMap<String, String> = HashMap()
+                    val random = (1111 until 9999).random()
+                    val myname = getname() + random
+                    usermap["name"] = myname
+                    usermap["email"] = email
+                    val user1 = it.result.user
+                    if (user1 != null) {
+                        usermap["uuid"] = it.result.user!!.uid
                     }
```

Comment:
1. Why `user1` and not just `user`?
2. `!!` risks a crash.  Don't deploy code with `!!` to your main branch.